File: common/src/main/java/dev/mayaqq/estrogen/features/dash/CommonDash.java
package dev.mayaqq.estrogen.features.dash;

import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

public class CommonDash {

    // dash is 5 game ticks = 250 ms; adding 50 ms for some leeway
    private static final long DASH_DURATION = 300;

    private static final ConcurrentHashMap<UUID, Long> players = new ConcurrentHashMap<>();

    public static void setDashing(UUID player) {
        players.put(player, System.currentTimeMillis());
    }

    public static void removeDashing(UUID player) {
        players.remove(player);
    }

    public static boolean isPlayerDashing(UUID player) {
        return players.getOrDefault(player, 0L) + DASH_DURATION > System.currentTimeMillis();
    }
}

File: src/main/kotlin/dev/mayaqq/estrogen/features/dash/CommonDash.kt
package dev.mayaqq.estrogen.features.dash

import java.util.*
import java.util.concurrent.ConcurrentHashMap

object CommonDash {
    private val DASH_DURATION = 300

    private val players = ConcurrentHashMap<UUID, Long>()

    fun setDashing(player: UUID) {
        players[player] = System.currentTimeMillis()
    }

    fun removeDashing(player: UUID) {
        players.remove(player)
    }

    fun isDashing(player: UUID): Boolean {
        return players.getOrDefault(player, 0L) + DASH_DURATION > System.currentTimeMillis()
    }
}